src/lib.rs

```diff
@@ -283,6 +283,12 @@ impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
             (_, true) => Self::MAX,
         }
     }
+
+    /// Check if this uint is zero
+    #[must_use]
+    pub fn is_zero(&self) -> bool {
+        self.limbs.iter().all(|&limb| limb == 0)
```

Comment:
Use `self == &Self::ZERO`. It's both simpler and allows the compiler to optimize better.